src/py21cmfast/wrapper/inputs.py

```diff
@@ -290,6 +300,102 @@ def from_subdict(cls, dct, safe=True):
         return cls.new(dct)
 
 
+@define(frozen=True, kw_only=True)
+class Table1D:
+    """Class for setting 1D interpolation table."""
+
+    size: int = field(converter=int, validator=validators.gt(0))
+    x_values: np.ndarray = field(
+        converter=lambda v: np.asarray(v, dtype=np.float64),
+        validator=validators.instance_of(np.ndarray),
+    )
+    y_values: np.ndarray = field(
+        converter=lambda v: np.asarray(v, dtype=np.float64),
+        validator=validators.instance_of(np.ndarray),
+    )
+
+    def __eq__(self, other):
+        """Determine if this is equal to another object."""
+        if self is None and other is None:
+            return True
+        elif (self is not None and other is None) or (
+            self is None and other is not None
+        ):
+            return False
+        else:
+            return (
+                self.size == other.size
+                and np.all(self.x_values == other.x_values)
+                and np.all(self.y_values == other.y_values)
+            )
```

Comment:
This is given to you for free by `attrs`, so you don't need to define it. You might have to pass `eq=attrs.cmp_using(eq=np.array_equal)` to the `field()` calls for the numpy arrays. See https://www.attrs.org/en/stable/comparison.html